engine/transaction/draft/outputs/op_return.go

```diff
@@ -0,0 +1,65 @@
+package outputs
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+
+	"github.com/bitcoin-sv/go-sdk/script"
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/transaction/errors"
+	"github.com/bitcoin-sv/spv-wallet/models/request/opreturn"
+)
+
+// OpReturn represents an OP_RETURN output specification.
+type OpReturn opreturn.Output
+
+func (o *OpReturn) evaluate(context.Context) (annotatedOutputs, error) {
+	if len(o.Data) == 0 {
+		return nil, txerrors.ErrDraftOpReturnDataRequired
+	}
+
+	data, err := o.getData()
+	if err != nil {
+		return nil, err
+	}
+	output, err := sdk.CreateOpReturnOutput(data)
+	if err != nil {
+		if errors.Is(err, script.ErrPartTooBig) {
+			return nil, txerrors.ErrDraftOpReturnDataTooLarge
+		}
+		return nil, spverrors.Wrapf(err, "failed to create OP_RETURN output")
+	}
+
+	annotation := transaction.NewDataOutputAnnotation()
+	return singleAnnotatedOutput(output, annotation), nil
+}
+
+func (o *OpReturn) getData() ([][]byte, error) {
+	data := make([][]byte, 0)
```

Comment:
You could preallocate the `data` slice (because the length of `o.Data` is known at this point)